src/ion/moves.rs

```diff
@@ -519,6 +502,73 @@ impl<'a, F: Function> Env<'a, F> {
                     }
                 }
 
+                // Scan debug-labels on this vreg that overlap with
+                // this range, producing a debug-info output record
+                // giving the allocation location for each label.
+                if !debug_labels.is_empty() {
+                    // Do a binary search to find the start of any
+                    // relevant labels. Recall that we require
+                    // debug-label requests to be sorted by vreg then
+                    // by inst-range.
+                    //
+                    // We search by the *end* of vale-label range as
+                    // the key in the slice (`tuple.2`), with the
+                    // *start* of this vreg range as the search key,
+                    // so we will find the first value-label range
+                    // with an end equal to or less than this range's
+                    // start. As long as we are careful to exclude the
+                    // just-adjacent case (end == start)
+                    // below, this will give us the first value-label
+                    // that possibly exists in this vreg range.
```

Comment:
I personally find it much easier to use `binary_search_by` and a predicate that never returns `Ordering::Equal`. Then you can just `unwrap_err` the result and get the index that is after the previous instruction but before the one that you are looking for.

Reply:
See the use of `binary_search_by` in `block_insts_and_edits` for an example of what I am describing.

Reply:
Switched to that style, thanks!